doc/source/development/rfc/rfc91_dataset_close.rst

```diff
@@ -0,0 +1,184 @@
+.. _rfc-91:
+
+=============================================================
+RFC 91: GDALDataset::Close() method
+=============================================================
+
+============== =============================================
+Author:        Even Rouault
+Contact:       even.rouault at spatialys.com
+Started:       2023-Jan-20
+Status:        Draft
+Target:        GDAL 3.7
+============== =============================================
+
+Summary
+-------
+
+This RFC aims at providing more robust error detection during dataset closing,
+in particular for datasets in creation/update mode.
+
+It modifies the GDALDataset::FlushCache() method and GDALFlushCache()
+function to return an error code, adds a GDALDataset::Close() virtual method and
+modifies GDALClose() to return an error code.
+
+Motivation
+----------
+
+A lot of drivers that have write capabilities may do write operations in their
+destructors, or just close the file descriptor owned by the dataset. Currently
+there is no clean way of catching errors that might occur. This recently bite
+`Fiona <https://github.com/Toblerity/Fiona/issues/1169>` where the GeoJSON driver
+may emit an error in its dataset destructor when writing the content of a file
+to a cloud object storage.
+
+Code that currently wants to detect if GDALClose() causes an error has to
+do something like the following:
+
+.. code-block:: c++
+
+    CPLErr eErrCodeBefore = CPLGetLastErrorType();
+    GDALClose(hOutDS);
+    if( eErrCodeBefore == CE_None && CPLGetLastErrorType() != CE_None )
+        printf("Error occurred during GDALClose()!\n")
+
+
+This is not pretty and there is always the slight risk that something in the
```

Comment:
suggestions: 
not pretty -> awkward
there is always the slight risk -> there is a risk

No need to qualify the risk (at least to me) as peoples' idea of level of risks might be different.

Reply:
done